### PR TITLE
Update Akka to 1.4.4

### PR DESCRIPTION
**Pull request by Arkatufus**
- Update to Akka ~~1.4.1-RC1~~ 1.4.4
- Use `common.props` instead of relying on hard coded version numbers.
- Implement Docker.DotNet to deploy Docker container for unit test validation.

### CODE REVIEW

**Comment by Arkatufus** [comment]
- Windows Redis is installed manually inside the windows build script.
- One test failing consistently. Need clarification: 
  When a journal entry with an event adapter that modifies its manifest is stored, will the stored journal manifest be equal to the original journal manifest? If not, then the unit test is broken.

**Comment by Arkatufus** [comment]
The code in question is this line:
https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Persistence.TCK/Serialization/JournalSerializationSpec.cs#L107

File: NuGet.Config [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <clear />
+    <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
```
Comment (Aaronontheweb) [inline NuGet.Config:5]: We can get rid of this - don't need it anymore.

File: build-system/azure-pipeline.template.yaml [hunk 2]
```diff
@@ -1,14 +1,12 @@
 parameters:
   name: ''
-  displayName: ''
```
Comment (Aaronontheweb) [inline build-system/azure-pipeline.template.yaml:3]: Roll back these changes - the current build files are fine and are in-use.

File: build-system/linux-pr-validation.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Pull request validation for Linux against the `dev` and `master` branches
```
Comment (Aaronontheweb) [inline build-system/linux-pr-validation.yaml:1]: Roll back these changes - the current build files are fine and are in-use.

File: build-system/windows-pr-validation.yaml [hunk 4]
```diff
@@ -14,17 +14,9 @@ pr:
 name: $(Year:yyyy).$(Month).$(DayOfMonth)$(Rev:.r)
 
 jobs:
-  - template: azure-pipeline.template.yaml
```
Comment (Aaronontheweb) [inline build-system/windows-pr-validation.yaml:17]: Roll back these changes - the current build files are fine and are in-use.

File: build-system/windows-release.yaml [hunk 5]
```diff
@@ -10,22 +10,23 @@ trigger:
...
+
 variables:
   - group: signingSecrets #create this group with SECRET variables `signingUsername` and `signingPassword`
   - group: nugetKeys #create this group with SECRET variables `nugetKey`
   - name: githubConnectionName
-    value: yourConnection #replace this
+    value: AkkaDotNet_Releases
   - name: projectName
-    value: yourProjectName #replace this
+    value: Akka.Persistence.SqlServer
```
Comment (Aaronontheweb) [inline build-system/windows-release.yaml:21]: Need to change this to Akka.Persistence.Redis

File: build.ps1 [hunk 6]
```diff
@@ -115,6 +116,18 @@ if (!(Test-Path $FakeExePath)) {
     }
 }
 
+###########################################################################
+# INSTALL REDIS
+###########################################################################
+$RedisExePath = Join-Path $ToolPath "redis-64/tools/redis-server.exe"
```
Comment (Aaronontheweb) [inline build.ps1:122]: Shouldn't need this if we're using Docker.NET

File: src/Akka.Persistence.Redis.Tests/RedisFixture.cs [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+using Xunit;
+
+namespace Akka.Persistence.Redis.Tests
+{
+    [CollectionDefinition("RedisSpec")]
+    public sealed class RedisSpecsFixture : ICollectionFixture<RedisFixture>
+    {
+    }
+
+    public class RedisFixture : IAsyncLifetime
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Redis.Tests/RedisFixture.cs:18]: LGTM

**Comment by Aaronontheweb** [comment]
@Arkatufus maybe we should run this on Linux only

**Comment by Arkatufus** [comment]
Agreed, this is very very annoying.

> 
> 
> @Arkatufus maybe we should run this on Linux only

Agreed, this is very very annoying.

**Pull request merged**